src/array.rs

```diff
@@ -462,12 +467,75 @@ impl<T: Element, D: Dimension> PyArray<T, D> {
             data_ptr as _,              // data
             mem::size_of::<T>() as i32, // itemsize
             0,                          // flag
-            ptr::null_mut(),            //obj
+            ptr::null_mut(),            // obj
         );
         PY_ARRAY_API.PyArray_SetBaseObject(ptr as *mut npyffi::PyArrayObject, cell as _);
         Self::from_owned_ptr(py, ptr)
     }
 
+    /// Creates a NumPy array backed by `array` the ownership of which is tied to the Python object `owner`
+    ///
+    /// # Safety
+    ///
+    /// `owner` is set as a base object for the return NumPy array, which means that `array` must not be dropped until `owner` is dropped.
+    /// Furthermore, `array` must not be reallocated after this method was called and until `owner` is dropped.
```

Comment:
```suggestion
    /// `owner` is set as a base object of the returned array, so it must not be dropped until `owner` is dropped.
    /// Furthermore, `array` must not be reallocated or destructed from the time this method is called and until `owner` is dropped.
```

Reply:
I think the "or destructed" part is redundant to the first sentence which already says that `array` must not be dropped/destructed before `owner`. Otherwise applied as suggested.